### PR TITLE
add my places view

### PR DESCRIPTION
**Pull request by dmikots**
Done my places view with view model. Add extensions to array. Add userDefault helper class.

### CODE REVIEW

File: AcloConcurrency/AcloConcurrency/AlcoConcurrencyApp.swift [hunk 1]
```diff
@@ -2,22 +2,18 @@ import SwiftUI
 
 @main
 struct AlcoConcurrencyApp: App {
-    
-    @StateObject private var collectionsViewModel: CollectionsViewModel
-    
+
+    @StateObject private var place: PlacesViewModel
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/AlcoConcurrencyApp.swift]: Add space between init and variables
And please use fullName placesViewModel. You can use(if you want) name just "viewModel" in your view or something, but not in Main Struct.

File: AcloConcurrency/AcloConcurrency/AlcoConcurrencyApp.swift [hunk 2]
```diff
@@ -2,22 +2,18 @@ import SwiftUI
...
     init() {
-        let filterService = FilterService()
-        _collectionsViewModel = StateObject(
-            wrappedValue: CollectionsViewModel(filterService: filterService)
-        )
+        _place = StateObject(wrappedValue: PlacesViewModel(userStorage: UserStorage()))
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/AlcoConcurrencyApp.swift]: hm, let use new variable 
`
let userStorage = UserStorage `
`
_place = StateObject(....(userStorage: userStorage))
`

File: AcloConcurrency/AcloConcurrency/Extensions/Array+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import Combine
+import Foundation
+
+extension Array: RawRepresentable where Element: Codable {
+    public init?(rawValue: String) {
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/Extensions/Array+Extensions.swift]: add space between extension and init

Comment (Mixorok) [reply AcloConcurrency/AcloConcurrency/Extensions/Array+Extensions.swift]: And use betterNaming like Array+UserDefaultsExtension

File: AcloConcurrency/AcloConcurrency/Helpers/UserStorage.swift [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import SwiftUI
+public class UserStorage: ObservableObject{
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/Helpers/UserStorage.swift]: add space between import and UserStorage
And I think ObservableObject useless here

File: AcloConcurrency/AcloConcurrency/Helpers/UserStorage.swift [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+    @AppStorage("places") var places: [PlaceModel] = [PlaceModel.mock]
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/Helpers/UserStorage.swift]: add space please
and use private var

File: AcloConcurrency/AcloConcurrency/MyPlaces/Models/PlaceModel.swift [hunk 6]
```diff
@@ -0,0 +1,18 @@
+//
+//  PlaceModel.swift
+//  AlcoConcurrency
+//
+//  Created by DariaMikots on 15.06.2022.
+//
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Models/PlaceModel.swift]: Delete

File: AcloConcurrency/AcloConcurrency/MyPlaces/Models/PlaceModel.swift [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+import Foundation
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Models/PlaceModel.swift:1]: Space

File: AcloConcurrency/AcloConcurrency/MyPlaces/Models/PlaceModel.swift [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+struct PlaceModel: Identifiable, Codable {
+    var id = UUID()
+    let name: String
+    let description: String
+    let image: String
+}
+
+extension PlaceModel {
+    static var mock = PlaceModel(id: UUID(), name: "Bar", description: "My favorite bar", image: "https://img.restoclub.ru/uploads/place/c/6/7/0/c670f3405f20b2c8513908b467548849_w958_h835--big.jpg")
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Models/PlaceModel.swift]: 100 digits

File: AcloConcurrency/AcloConcurrency/MyPlaces/ViewModel/PlacesViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        Task{
+            self.places = userStorage.places
+        }
+    }
+    @MainActor func savePlace(_ place: PlaceModel) {
+        Task{
+            userStorage.saveNewPlace(place)
+        }
+        getPlaces()
+    }
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/ViewModel/PlacesViewModel.swift]: Right now we don't need here MainActor and Task. You can remove and later we think about how to use async

File: AcloConcurrency/AcloConcurrency/MyPlaces/Views/AddCustomCollectionSheet.swift [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+struct AddCustomCollectionSheet: View {
+    @ObservedObject private var placeModel: PlacesViewModel
+    init(placeModel: PlacesViewModel) {
+        self.placeModel = placeModel
+    }
+    @State private var name: String = ""
+    @State private var description: String = ""
+    @Environment(\.presentationMode) var presentationMode
+    var body: some View {
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Views/AddCustomCollectionSheet.swift]: use spaces

1. Variables
2. Init
3. func(var body)
 and spaces!

File: AcloConcurrency/AcloConcurrency/MyPlaces/Views/MyPlaceRow.swift [hunk 11]
```diff
@@ -0,0 +1,43 @@
+import SwiftUI
+
+struct MyPlaceRow: View {
+    var place: PlaceModel
+    var body: some View {
+        HStack{
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Views/MyPlaceRow.swift]: use private var place 
init()
spaces

File: AcloConcurrency/AcloConcurrency/MyPlaces/Views/PlacesView.swift [hunk 12]
```diff
@@ -0,0 +1,50 @@
+import SwiftUI
+
+struct PlacesView: View {
+
+    @ObservedObject private var placeModel: PlacesViewModel
+
+    init(placeModel: PlacesViewModel) {
+        self.placeModel = placeModel
+    }
+    @State private var showingSheet = false
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Views/PlacesView.swift]: move up

File: AcloConcurrency/AcloConcurrency/AlcoConcurrencyApp.swift (hunk 1 above)
Comment (dmikots) [reply AcloConcurrency/AcloConcurrency/AlcoConcurrencyApp.swift]: how can I get fck linter ....

File: AcloConcurrency/AcloConcurrency/Helpers/UserStorage.swift [hunk 13]
```diff
@@ -1,7 +1,11 @@
 import SwiftUI
 public class UserStorage: ObservableObject{
-    @AppStorage("places") var places: [PlaceModel] = [PlaceModel.mock]
-    func saveNewPlace(_ newPlace: PlaceModel)  {
+
+    @AppStorage("places") private var places: [PlaceModel] = [PlaceModel.mock]
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/Helpers/UserStorage.swift:5]: space

File: AcloConcurrency/AcloConcurrency/Helpers/UserStorage.swift [hunk 14]
```diff
@@ -1,7 +1,11 @@
...
+    func saveNewPlace(_ newPlace: PlaceModel) {
         places.append(newPlace)
-        }
     }
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/Helpers/UserStorage.swift:9]: also space

File: AcloConcurrency/AcloConcurrency/MyPlaces/Views/AddCustomCollectionSheet.swift [hunk 15]
```diff
@@ -1,21 +1,31 @@
...
+
     @ObservedObject private var placeModel: PlacesViewModel
+
     init(placeModel: PlacesViewModel) {
         self.placeModel = placeModel
     }
+
     @State private var name: String = ""
     @State private var description: String = ""
     @Environment(\.presentationMode) var presentationMode
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Views/AddCustomCollectionSheet.swift]: move up to ObserObj...

File: AcloConcurrency/AcloConcurrency/MyPlaces/Views/MyPlaceRow.swift [hunk 16]
```diff
@@ -1,7 +1,12 @@
 import SwiftUI
 
 struct MyPlaceRow: View {
-    var place: PlaceModel
+    private var place: PlaceModel
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Views/MyPlaceRow.swift:5]: space

File: AcloConcurrency/AcloConcurrency/MyPlaces/Views/AddCustomCollectionSheet.swift [hunk 17]
```diff
@@ -1,21 +1,31 @@
 import SwiftUI
 
 struct AddCustomCollectionSheet: View {
+
     @ObservedObject private var placeModel: PlacesViewModel
+
     init(placeModel: PlacesViewModel) {
```
Comment (MaksimBezdrobnoi) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Views/AddCustomCollectionSheet.swift]: please don't injection all viewModel. Add only functions which you want

File: AcloConcurrency/AcloConcurrency/MyPlaces/Models/PlaceModel.swift [hunk 18]
```diff
@@ -1,18 +1,13 @@
...
-
 import Foundation
+
 struct PlaceModel: Identifiable, Codable {
     var id = UUID()
     let name: String
     let description: String
-    let image: String
+    var image: String =
+"https://img.restoclub.ru/uploads/place/c/6/7/0/c670f3405f20b2c8513908b467548849_w958_h835--big.jpg"
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Models/PlaceModel.swift]: first let
second var

File: AcloConcurrency/AcloConcurrency/MyPlaces/Views/AddCustomCollectionSheet.swift [hunk 19]
```diff
@@ -1,21 +1,26 @@
...
-        self.placeModel = placeModel
-    }
+
+    @Environment(\.presentationMode) var presentationMode
+
     @State private var name: String = ""
     @State private var description: String = ""
-    @Environment(\.presentationMode) var presentationMode
+
+    private let savePlace: (_ place: PlaceModel) -> Void
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Views/AddCustomCollectionSheet.swift]: first let
second var

File: AcloConcurrency/AcloConcurrency/MyPlaces/Views/AddCustomCollectionSheet.swift [hunk 20]
```diff
@@ -1,21 +1,26 @@
 import SwiftUI
 
 struct AddCustomCollectionSheet: View {
-    @ObservedObject private var placeModel: PlacesViewModel
-    init(placeModel: PlacesViewModel) {
-        self.placeModel = placeModel
-    }
+
+    @Environment(\.presentationMode) var presentationMode
```
Comment (Mixorok) [inline AcloConcurrency/AcloConcurrency/MyPlaces/Views/AddCustomCollectionSheet.swift]: private var

**Review by Mixorok — APPROVED** [review]

**Pull request merged**